src/vtbackend/Terminal.cpp

```diff
@@ -1253,8 +1253,6 @@ namespace
 
 string Terminal::extractSelectionText() const
 {
-    auto const _ = std::scoped_lock { *this };
```

Comment:
Wen removed here, we need to add one from outside here:

```cpp
bool TerminalSession::operator()(actions::CopySelection copySelection);
bool TerminalSession::operator()(actions::OpenSelection);
```

All the other cases I found were indirectly governed by the caller (/of a caller) already.